packages/plugin-core/src/test/suite-integ/NoteLookupCommand.test.ts

```diff
@@ -671,5 +713,611 @@ suite("NoteLookupCommand", function () {
+        },
+        onInit: async ({ wsRoot, vaults, engine }) => {
+          const { cmdOut, selectedText } = await prepareCommandFunc({
+            vaults, 
+            engine, 
+            noteType: LookupNoteTypeEnum.scratch 
+          });
+
+          const newNote = NoteUtils.getNoteOrThrow({
+            fname: cmdOut!.quickpick.value,
+            notes: engine.notes,
+            vault: vaults[0],
+            wsRoot,
+          });
+          expect(newNote.body.trim()).toEqual(selectedText);
+
+          done();
+        }
+      });
+    });
+  });
+
+  describe("multiselect interactions", () => {
+    // TODO: there's gotta be a better way to mock this.
+    const prepareCommandFunc = async ({ wsRoot, vaults, engine, opts }: any) => {
+      const cmd = new NoteLookupCommand();
+      const notesToSelect = ["foo.ch1", "bar", "lorem", "ipsum"].map((fname) => engine.notes[fname]);
+      const selectedItems = notesToSelect.map((note) => {
+        return DNodeUtils.enhancePropForQuickInputV3({
+          props: note,
+          schemas: engine.schemas,
+          wsRoot,
+          vaults
+        });
+      }) as NoteQuickInput[];
+
+      const runOpts = {
+        multiselect: true,
+        noConfirm: true
+      } as CommandRunOpts;
+
+      if (opts.split) runOpts.splitType = LookupSplitTypeEnum.horizontal;
+      
+      const gatherOut = await cmd.gatherInputs(runOpts);
+
+      const mockQuickPick = createMockQuickPick({
+        value: "foo.ch1",
+        selectedItems,
+        canSelectMany: true,
+        buttons: gatherOut.quickpick.buttons,
+      });
+
+      if (opts.copyLink) {
+        mockQuickPick.show();
+        const { copyNoteLinkBtn } = getEffectTypeButtons(mockQuickPick.buttons);
+        await gatherOut.controller.onTriggerButton(copyNoteLinkBtn);
+        const content = await clipboard.readText();
+        return { content };
+      };
+
+      mockQuickPick.showNote = gatherOut.quickpick.showNote;
+      
+      sinon.stub(cmd, "enrichInputs").returns(
+        Promise.resolve({
+          quickpick: mockQuickPick,
+          controller: gatherOut.controller,
+          provider: gatherOut.provider, 
+          selectedItems
+        })
+      );
+      return { cmd };
+    } 
+    
+    test("split + multiselect: should have n+1 columns", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+          await NoteTestUtilsV4.createNote({
+            fname: "lorem",
+            vault: vaults[0],
+            wsRoot
+          });
+          await NoteTestUtilsV4.createNote({
+            fname: "ipsum",
+            vault: vaults[0],
+            wsRoot
+          });
+        },
+        onInit: async ({ wsRoot, vaults, engine }) => {
+          // make clean slate.
+          VSCodeUtils.closeAllEditors();
+
+          VSCodeUtils.openNote(engine.notes["foo"]);
```

Comment:
if you add an `await` here, you should be able to remove the `setTimeout`

Reply:
ooooh I feel dumb now hahaha